src/pages/Trends/connectedWordsEpic.js

```diff
@@ -0,0 +1,234 @@
+import { Observable } from 'rxjs'
+import { handleErrorAndTriggerAction } from '../../epics/utils'
+import {
+  TRENDS_HYPED_FETCH_SUCCESS,
+  TRENDS_HYPED_FETCH_TICKERS_SLUGS_SUCCESS,
+  TRENDS_CONNECTED_WORDS_SUCCESS,
+  TRENDS_CONNECTED_WORDS_FAILED,
+  TRENDS_CONNECTED_WORDS_OPTIMIZATION_SUCCESS
+} from '../../components/Trends/actions'
+import { ALL_INSIGHTS_BY_TAG_QUERY } from '../../components/Insight/insightsGQL'
+import { binarySearch } from './utils'
+
+const oneDayTimestamp = 1000 * 60 * 60 * 24
+
+const getInsightTrendTagByDate = date =>
+  `${date.getDate()}-${date.getMonth()}-${date.getFullYear()}-trending-words`
+
+const tickerCheckFn = (target, { ticker }) => target === ticker
+const tickerMoveFn = (target, { ticker }) => target < ticker
+const slugCheckFn = (target, { slug }) => target === slug
+const slugMoveFn = (target, { slug }) => target < slug
+
+let projectsSortedByTicker
+let projectsSortedBySlug
+
+const basicSort = (a, b) => {
+  if (a > b) {
+    return 1
+  }
+  if (a < b) {
+    return -1
+  }
+  return 0
+}
+
+const mapWordToProjectsTicker = word => {
+  const { value: { ticker: ticker1 } = {} } = binarySearch({
+    array: projectsSortedByTicker,
+    target: word,
+    checkFn: tickerCheckFn,
+    moveFn: tickerMoveFn
+  })
+
+  if (ticker1) {
+    return ticker1
+  }
+
+  const { value: { ticker: ticker2 } = {} } = binarySearch({
+    array: projectsSortedBySlug,
+    target: word,
+    checkFn: slugCheckFn,
+    moveFn: slugMoveFn
+  })
+
+  if (ticker2) {
+    return ticker2
+  }
+
+  const { ticker: ticker3 } =
+    projectsSortedByTicker.find(({ name }) => name.includes(word)) || {}
+
+  return ticker3
+}
+
+export const connectedWordsOptimizationEpic = action$ =>
+  action$
+    .ofType(TRENDS_HYPED_FETCH_TICKERS_SLUGS_SUCCESS)
+    .take(1)
+    .switchMap(({ payload: { allAssets } }) => {
+      const { length } = allAssets
+      const tempAssets = allAssets.slice()
+
+      for (let i = 0; i < length; i++) {
```

Comment:
why not map? @DmitriVanGuard

Reply:
Slower for an array of 1000+ elements with an allocation memory for a new object on every iteration